src/ORM/DataList.php

```diff
@@ -1308,13 +1316,28 @@ private function fetchEagerLoadManyMany(
         $relationListClass = get_class($relationList);
 
         // Get ALL of the items for this relation up front, for ALL of the parents
-        $fetchedRows = $relationList->forForeignID($parentIDs)->getFinalisedQuery();
+        $fetchList = $relationList->forForeignID($parentIDs);
+        $fetchList = $this->manipulateEagerLoadingQuery($fetchList, $relationChain, $relationType);
+        $fetchedRows = $fetchList->getFinalisedQuery();
 
         foreach ($fetchedRows as $row) {
             $fetchedRowsArray[$row['ID']] = $row;
             $fetchedIDs[] = $row['ID'];
         }
 
+        // Get the join records so we can correctly identify which children belong to which parents
+        // This also holds extra fields data
+        $fetchedIDsAsString = implode(',', $fetchedIDs);
+        $joinRows = DB::query(
+            'SELECT * FROM "' . $joinTable
+            // Only get joins relevant for the parent list
+            . '" WHERE "' . $parentIDField . '" IN (' . implode(',', $parentIDs) . ')'
+            // Exclude any children that got filtered out
+            . ' AND ' . $childIDField . ' IN (' . $fetchedIDsAsString . ')'
+            // Respect sort order of fetched items
+            . ' ORDER BY FIELD(' . $childIDField . ', ' . $fetchedIDsAsString . ')'
```

Comment:
The sorting is important here because the join rows are what we're looping through - so the order of these join rows determines the order of the records in the resultant `EagerLoadedList`